### PR TITLE
Login UI and project setup

### PR DESCRIPTION
**Pull request by danialepaco**
Login UI 
Home tabs
Project auth setup

Simulator Screenshot - iPhone 15 Pro Max - 2024-06-25 at 16 39 12

### CODE REVIEW

File: app/login.tsx [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+export default function Login() {
+  const { signIn } = useSession();
+  const handleLogin = () => {
+    signIn();
+    router.replace("/");
+  };
+
+  return (
+    <ImageBackground source={require('../assets/images/background.jpg')} resizeMode="cover" style={styles.background}>
```
Comment (juanchoperezj) [inline app/login.tsx]: Not a blocker for this PR, there is an approach to define all images/icons files from one file and use them from there, e.g.:
```
export const background = require()
// other images
```

File: app/login.tsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+          <TouchableOpacity style={styles.registerButton}>
+              <Text style={styles.registerButtonTitle}>Sign up</Text>
+            </TouchableOpacity>
+        </View>
+      </View>
+    </ImageBackground>
+  );
+}
+
+const styles = StyleSheet.create({
```
Comment (juanchoperezj) [inline app/login.tsx]: To keep files shorter, styles are usually placed in separate files.

File: components/Themed.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+export function Text(props: TextProps) {
+  const { style, lightColor, darkColor, ...otherProps } = props;
+  const color = useThemeColor({ light: lightColor, dark: darkColor }, 'text');
+
+  return <DefaultText style={[{ color }, style]} {...otherProps} />;
+}
+
+export function View(props: ViewProps) {
```
Comment (juanchoperezj) [inline components/Themed.tsx:40]: If you add more components here, consider moving them into separate files.

**Review by juanchoperezj — APPROVED** [review]

File: app/login.tsx (hunk 1 above)
Comment (fernandatoledo) [reply app/login.tsx]: Something like this: https://github.com/rootstrap/driven-react-native/blob/develop/src/assets/images/index.tsx

File: app/(auth)/(tabs)/_layout.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    <Tabs
+      screenOptions={{
+        tabBarActiveTintColor: Colors[colorScheme ?? "light"].tint,
+        headerShown: useClientOnlyValue(false, true),
+      }}
+    >
+      <Tabs.Screen
+        name="index"
+        options={{
+          title: "Tab One",
```
Comment (fernandatoledo) [inline app/(auth)/(tabs)/_layout.tsx:29]: Is localization part of a future PR? if that's the case can you add a todo in this screen to move literal strings to the locales?

File: app/(auth)/(tabs)/index.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        onPress={() => {
+          // The `app/(app)/_layout.tsx` will redirect to the sign-in screen.
+          signOut();
+        }}
+      />
+    </View>
+  );
+}
+
+const styles = StyleSheet.create({
```
Comment (fernandatoledo) [inline app/(auth)/(tabs)/index.tsx]: Please keep styles in a separate file. This helps maintain a clear separation of concerns, making the code more maintainable and readable.

File: app/(auth)/(tabs)/_layout.tsx (hunk 4 above)
Comment (fernandatoledo) [reply app/(auth)/(tabs)/_layout.tsx:29]: After checking the current backlog of blackmarket I don't think we have a ticket for it but I will add it for future blackmarkets 😅 Can you add it to your own backlog? :)

File: app/(auth)/modal.tsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+import { Platform, StyleSheet } from 'react-native';
+
+import EditScreenInfo from '@/components/EditScreenInfo';
+import { Text, View } from '@/components/Themed';
+
+export default function ModalScreen() {
+  return (
+    <View style={styles.container}>
+      <Text style={styles.title}>Modal</Text>
+      <View style={styles.separator} lightColor="#eee" darkColor="rgba(255,255,255,0.1)" />
```
Comment (fernandatoledo) [inline app/(auth)/modal.tsx]: It's also a good practice to have all colors in a constants file. Generally, the same colors are repeatedly used across different screens, so having them in constants makes it easier to maintain.

File: app/+html.tsx
Comment (fernandatoledo) [inline app/+html.tsx:1]: I don't think this file should be used for React Native, honestly I don't have experience with expo-router so please double check but from what I understand of the [docs](https://docs.expo.dev/router/reference/static-rendering/#root-html) it doesn't apply

File: app/_layout.tsx [hunk 7]
```diff
@@ -26,12 +38,14 @@ export default function RootLayout() {
     return null;
   }
 
+  return <RootLayoutNav />;
+}
+import { Slot } from "expo-router";
```
Comment (fernandatoledo) [inline app/_layout.tsx]: Please keep all your imports at the top of the file and avoid having more than one component per file

File: app/ctx.tsx [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+  isLoading: boolean;
+}>({
+  signIn: () => null,
+  signOut: () => null,
+  session: null,
+  isLoading: false,
+});
+
+// This hook can be used to access the user info.
+export function useSession() {
```
Comment (fernandatoledo) [inline app/ctx.tsx]: Please move this hook to a hooks folder maybe inside the auth folder

File: app/login.tsx (hunk 1 above)
Comment (danialepaco) [reply app/login.tsx]: Thanks!

File: components/Themed.tsx (hunk 3 above)
Comment (danialepaco) [reply components/Themed.tsx:40]: It is something expo added me on the template

File: app/(auth)/(tabs)/_layout.tsx (hunk 4 above)
Comment (danialepaco) [reply app/(auth)/(tabs)/_layout.tsx:29]: Sure I can add it!

File: app/(auth)/(tabs)/index.tsx (hunk 5 above)
Comment (danialepaco) [reply app/(auth)/(tabs)/index.tsx]: Separating this from the index folder is breaking the expo-router. How should I create a separate file?

Comment (fernandatoledo) [reply app/(auth)/(tabs)/index.tsx]: You can create a styles.js file on app/(auth)/(tabs)/styles.js

```
import {  StyleSheet } from "react-native";

export default styles = StyleSheet.create({
    container: {
      flex: 1,
      alignItems: "center",
      justifyContent: "center",
    },
    title: {
      fontSize: 20,
      fontWeight: "bold",
    },
    separator: {
      marginVertical: 30,
      height: 1,
      width: "80%",
    },
  });
```

And then on index.tsx for example you can import your styles from the styles.js file 

`import styles from './styles';`

Comment (danialepaco) [reply app/(auth)/(tabs)/index.tsx]: thanks!

**Pull request merged**